core/server/api/v2/settings.js

```diff
@@ -28,10 +29,15 @@ module.exports = {
             // CASE: omit core settings unless internal request
             if (!frame.options.context.internal) {
                 settings = _.filter(settings, (setting) => {
-                    return setting.type !== 'core' && setting.key !== 'permalinks';
+                    return setting.type !== 'core';
                 });
             }
 
+            // Always filter out deprecated settings
+            settings = _.filter(settings, (setting) => {
+                return !_.includes(deprecatedSettings, setting.key);
+            });
```

Comment:
This piece of logic belongs IMO to the output serializer. We remove fields from the output for v2.

Future:
If we have to maintain 3 API versions at a time, it might be helpful if the controller query implementation stays as generic/general as possible to be able to re-use the query logic e.g. either outsource to shared or copy to a new API version.

Open for discussion 👍

Reply:
I have only updated the logic where it was in the codebase already, as it only removed permalinks and only sometimes.

I can move it, but that requires me to touch this code: https://github.com/TryGhost/Ghost/blob/2d7def71341e1b6d5ccc73b65ba28cb6f298b1ee/core/server/api/v2/utils/serializers/output/settings.js#L14

And I can't quite follow why the filter is so complex

Reply:
👍 looks good now